### PR TITLE
[Merged by Bors] - chore: delete `Init.Data.Fin.Basic`

### PR DESCRIPTION
**Pull request by Parcly-Taxel**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Init/Data/Fin/Basic.lean [hunk 1]
```diff
@@ -1,34 +0,0 @@
...
-variable {n : ℕ} {i j : Fin n}
-
-@[deprecated eq_of_val_eq (since := "2024-02-15")]
-theorem eq_of_veq : i.val = j.val → i = j := eq_of_val_eq
-
-@[deprecated val_eq_of_eq (since := "2024-02-15")]
-theorem veq_of_eq : i = j → i.val = j.val := val_eq_of_eq
-
--- These two aren't deprecated because `ne_of_val_ne` and `val_ne_of_ne`
--- use `¬a = b` instead of `a ≠ b`. TODO: fix or rename in Lean core.
```
Comment (jcommelin) [inline Mathlib/Init/Data/Fin/Basic.lean:30]: This comment has not been addressed. But seemingly this syntactic difference does not cause any problems in mathlib.

File: Mathlib/Data/List/Nodup.lean [hunk 2]
```diff
@@ -3,9 +3,9 @@ Copyright (c) 2018 Mario Carneiro. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Mario Carneiro, Kenny Lau
 -/
+import Mathlib.Data.Fin.Basic
```
Comment (jcommelin) [inline Mathlib/Data/List/Nodup.lean:6]: Why do you need this import? The init file that was imported before only has deprecated stuff in it. So this file should be able to work with the core versions, right?

Comment (Parcly-Taxel) [reply Mathlib/Data/List/Nodup.lean]: Fixed.

File: Mathlib/Init/Data/Fin/Basic.lean (hunk 1 above)
Comment (eric-wieser) [reply Mathlib/Init/Data/Fin/Basic.lean:30]: Can we open a PR to core that changes these two lemmas to use `≠`? That seems uncontroversial to me.

Comment (Parcly-Taxel) [reply Mathlib/Init/Data/Fin/Basic.lean:30]: @eric-wieser https://github.com/leanprover/lean4/pull/5011

Comment (semorrison) [reply Mathlib/Init/Data/Fin/Basic.lean:30]: I've merged this now.

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**